docs/source/concepts/local-ledger.rst

```diff
@@ -0,0 +1,452 @@
+.. Copyright (c) 2020 The DAML Authors. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+.. _local-ledger:
+
+Causality and Local Ledgers
+###########################
+
+DAML ledgers do not totally order all transactions.
+So different parties may observe two transactions on different Participant Nodes in different orders via the :ref:`Ledger API <ledger-api-services>`.
+Moreover, different Participant Nodes may output two transactions for the same party in different orders.
+This document explains the ordering guarantees that DAML ledgers do provide, by :ref:`example <causality-examples>` and formally via the concept of :ref:`causality graphs <causality-graph>` and :ref:`local ledgers <local-ledger-structure>`.
+
+The presentation assumes that you are familiar with the following concepts:
+
+* The :ref:`Ledger API <ledger-api-services>`
+
+* The :ref:`DAML Ledger Model <da-ledgers>`
+
+.. _causality-examples:
+
+Causality examples
+******************
+
+The following examples illustrate the ordering guarantees of the Ledger API.
+They use the paint counteroffer workflow from the DAML Ledger Model's :ref:`privacy section <da-model-privacy>`.
+Recall that :ref:`the party projections <da-paint-counteroffer-example>` are as follows:
+
+.. https://www.lucidchart.com/documents/edit/c4df0455-13ab-415f-b457-f5654c2684be
+.. image:: ./ledger-model/images/divulgence-for-disclosure-counteroffer.svg
+   :align: center
+   :width: 100%
+
+
+.. _causality-example-create-archive:
+
+Stakeholders of a contract see creation and archival in the same order.
+=======================================================================
+
+The creation of the `CounterOffer A P Bank` is ordered before the painter exercising the consuming choice on the `CounterOffer`.
+(If the **Create** was ordered after the **Exercise**, the resulting shared ledger would be inconsistent, which violates the validity guarantee of DAML ledgers.)
+Accordingly, Alice will see the creation before the archival on her transaction stream and so will the painter.
+This does not depend on whether they are hosted on the same Participant Node.
+
+.. _causality-example-create-use-archive:
+
+Signatories of a contract and stakeholder actors see usages after the creation and before the archival.
+=======================================================================================================
+
+The `Fetch A (Iou Bank A)` action comes after the creation of the `Iou Bank A` and before its archival,
+for both Alice and the Bank,
+because the Bank is a signatory of the `Iou Bank A` contract and Alice is stakeholder of the `Iou Bank A` contract and an actor on the **Fetch** action.
```

Comment:
```suggestion
because the Bank is a signatory of the `Iou Bank A` contract and Alice is a stakeholder of the `Iou Bank A` contract and an actor on the **Fetch** action.
```